R/filter_pd.R

```diff
@@ -0,0 +1,99 @@
+#' Filter a Dataset to Only Include the Source Parameter Records up to and
+#' including the first PD. These records are passed to downstream derivations
+#' regarding responses such as BOR.
+#'
+#' @param dataset An input BDS style data set is expected
+#'
+#' @param filter_source A filter condition is expected, upon which the ADRS
+#'        data set will be filtered to capture the PARAM containing the PD
+#'        response
+#'
+#' @param source_pd The variable name which contains the dates of the responses
+#'        A date/datetime object is expected.
+#'
+#' @param source_datasets A named list of data sets is expected.
+#'        Default is ADRS.
+#'
+#' @return A subset data set of ADRS of records that only include
+#'         the source_param records, keeping only those occurring
+#'        up to and including the first PD,
+#'         where the date comes from the ADT of
+#'         the source_param/source_pd records.
```

Comment:
replace with "where the date comes from the `source_pd` date source" - as it doesn't necessarily need to be ADT. the user could use ADSL.PDDT for example